src/main/java/com/github/jasminb/jsonapi/JSONAPIDocument.java

```diff
@@ -93,10 +93,17 @@ public T get() {
 	 *
 	 * @param meta {@link Map} meta
 	 */
-	public void setMeta(Map<String, ?> meta) {
+	public void setMeta(Map<String, Object> meta) {
 		this.meta = new HashMap<>(meta);
 	}
 
+	public void addMeta(String key, Object value) {
```

Comment:
This can be left as it is and provides intended benefit (adding arbitrary meta without having to worry about creating the container first).

Reply:
You cannot put the `Object` value in a `Map<String, ?>`.

Reply:
See above comment,  the first change is valid, it needs to be `Map<String, Object`, however, existing method signatures should be kept (which is compatible with the change). Helper method adds convinence that you intended and it can have `<String, Object>` signature.

Reply:
OK will have a look later today!